test/functional/test_framework/test_framework.py

```diff
@@ -821,6 +822,29 @@ def skip_if_no_py3_zmq(self):
         except ImportError:
             raise SkipTest("python3-zmq module not available.")
 
+    def skip_if_no_python_bcc(self):
+        """Attempt to import the bcc package and skip the tests if the import fails."""
+        try:
+            import bcc  # type: ignore[import] # noqa: F401
+        except ImportError:
+            raise SkipTest("bcc python module not available")
+
+    def skip_if_no_bitcoind_tracepoints(self):
+        """Skip the running test if bitcoind has not been compiled with USDT tracepoint support."""
+        if not self.is_usdt_compiled():
+            raise SkipTest("bitcoind has not been built with USDT tracepoints enabled.")
+
+    def skip_if_no_bpf_permissions(self):
+        """Skip the running test if we don't have permissions to do BPF syscalls and load BPF maps."""
+        # check for 'root' permissions
+        if os.geteuid() != 0:
+            raise SkipTest("no permissions to use BPF (please review the tests carefully before running them with higher privileges)")
```

Comment:
As alternative to / in addition to `os.geteuid() != 0` for non-root use we'll probably want to check `CapEff` in `/proc/<pid>/status` for `(1 << CAP_PERFMON) | (1 << CAP_BPF)`.
```
CAP_PERFMON = 38
CAP_BPF = 39
```
Not necessarily in this PR though.